platforms/documentation/docs/src/docs/userguide/reference/gradle_daemon.adoc

```diff
@@ -180,20 +180,23 @@ Gradle defaults to the current shell path and `JAVA_HOME` environment variable t
 
 Alternatively, a different JVM installation can be specified for the build using the <<build_environment.adoc#sec:gradle_configuration_properties,`org.gradle.java.home` Gradle property>> or programmatically through the Tooling API.
 
-If Daemon JVM criteria is available, it takes precedence over `JAVA_HOME` and `org.gradle.java.home`.
-
 Building on the <<toolchains.adoc#toolchains,toolchain feature>>, you can now use declarative criteria to specify the JVM requirements for the build.
 
 === Daemon JVM criteria
 
-The _daemon JVM criteria_ is controlled by a task, similarly to how <<gradle_wrapper.adoc#customizing_wrapper,`wrapper` task>> updates the wrapper properties.
+The _Daemon JVM criteria_ is controlled by a task, similar to how the <<gradle_wrapper.adoc#customizing_wrapper, `wrapper` task>> updates the wrapper properties file.
 When the task runs, it creates or updates the criteria in the `gradle/gradle-daemon-jvm.properties` file.
 For more control, the task can be further configured in the build script or via command-line arguments.
 
-As with the wrapper, the generated file should be checked into version control.
-This will ensure any developer or CI server that runs the build will use the same JVM version.
+If the Daemon JVM criteria configuration is provided, it takes precedence over `JAVA_HOME` and `org.gradle.java.home`.
+
+WARNING: With the addition of auto-provisioning support, the default setup requires that <<toolchains.adoc#sub:download_repositories, toolchain download repositories>> be configured.
```

Comment:
The [toolchains.adoc#sub:download_repositories, toolchain download repositories](https://docs.gradle.org/current/userguide/toolchains.html#sub:download_repositories) link suggest the `Foojay plugin` as an example but probably will be worth mentioning this is the recommended and supprted by Gradle. However, for more advance configurations a custom plugin should be specified
```
One example of a toolchain resolver plugin is the [Foojay Toolchains Plugin](https://github.com/gradle/foojay-toolchains), based on the [foojay Disco API](https://github.com/foojayio/discoapi). It even has a convention variant, which automatically takes care of all the needed configuration, just by being applied:
```

Reply:
Fixed.

Reply:
Fixed.